includes/orders/functions/actions.php

```diff
@@ -71,29 +71,39 @@ function edd_add_manual_order( $args = array() ) {
 	// Create a new customer record.
 	if ( isset( $order_data['edd-new-customer'] ) && 1 === absint( $order_data['edd-new-customer'] ) ) {
```

Comment:
This is not technically part of this issue, but I experimented with selecting "add a new customer" and then entering an email address for an existing customer and that was kind of fun:

```
WordPress database error Duplicate entry 'testpaypal@edd.local' for key 'email' for query INSERT INTO `wp_edd_customers` (`user_id`, `email`, `name`, `status`, `purchase_value`, `purchase_count`, `date_created`, `date_modified`, `uuid`) VALUES (241, 'testpaypal@edd.local', 'Test PayPal', 'active', '0', '0', '2022-08-19 13:48:42', '2022-08-19 13:48:42', 'urn:uuid:d8d0609e-ed10-4492-bde2-3a68ebe57759') made by require_once('wp-admin/admin.php'), require_once('wp-load.php'), require_once('wp-config.php'), require_once('wp-settings.php'), do_action('init'), WP_Hook->do_action, WP_Hook->apply_filters, edd_post_actions, do_action('edd_add_order'), WP_Hook->do_action, WP_Hook->apply_filters, edd_add_manual_order, edd_add_customer, EDD\Database\Query->add_item
[19-Aug-2022 13:48:42 UTC] PHP Warning:  Attempt to read property "id" on bool in \easy-digital-downloads\includes\orders\functions\actions.php on line 241
```

So I'm feeling like we should harden this check to make sure that a customer doesn't already exist.

Otherwise this is working correctly, so if you want to handle that in a separate issue then I'm okay with that.

Reply:
@robincornett I can harden this, to that if it finds a customer with that email address it attaches it to the existing customer instead.

We can open an issue to improve the UI to do a customer check and suggest the existing customer if it's found.

Reply:
@robincornett Take a look at these latest changes to see if you like this. I had to update the `edd_get_customer_by` function to account for non-primary email addresses.

If you think this is too much for 3.0.4 we can roll it back and just look for primary email addresses as well.